File: src/org/webfabric/web/sitemesh/StringTemplateDecorator.java
package org.webfabric.web.sitemesh;

import com.opensymphony.module.sitemesh.HTMLPage;
import org.antlr.stringtemplate.StringTemplate;
import org.antlr.stringtemplate.NoIndentWriter;

import java.io.Writer;
import java.io.IOException;

public class StringTemplateDecorator {
    private final StringTemplate template;

    public StringTemplateDecorator(StringTemplate template) {
        this.template = template;
    }

    public void Decorate(HTMLPage html, Writer writer) throws IOException {
        template.setAttribute("page", html);
        template.setAttribute("head", html != null ? html.getHead() : null );
        template.setAttribute("title", html != null ? html.getTitle() : null);
        template.setAttribute("body", html != null ? html.getBody() : null);
        template.write(new NoIndentWriter(writer));
    }
}

File: src/org/webfabric/web/sitemesh/StringTemplateDecorator.scala
package org.webfabric.web.sitemesh

import _root_.org.webfabric.web.servlet.{QueryString, ContextPath}
import com.opensymphony.module.sitemesh.HTMLPage
import java.io.{Writer, StringWriter}
import org.antlr.stringtemplate.{StringTemplate, NoIndentWriter}

class StringTemplateDecorator(template: StringTemplate) {
  def setBase(base: ContextPath): StringTemplateDecorator = {
    template.setAttribute("base", base)
    this
  }

  def setQueryString(queryString: QueryString): StringTemplateDecorator = {
    template.setAttribute("query", queryString.map)
    this
  }

  def setPage(page: HTMLPage): StringTemplateDecorator = {
    template.setAttribute("page", page)
    template.setAttribute("head", page.getHead)
    template.setAttribute("title", page.getTitle)
    template.setAttribute("body", page.getBody)
    this
  }

  def writeTo(writer: Writer) = template.write(new NoIndentWriter(writer))

  override def toString = {
    val writer = new StringWriter
    writeTo(writer)
    writer.toString
  }
}